mono/sgen/sgen-gc.c

```diff
@@ -326,6 +326,15 @@ nursery_canaries_enabled (void)
 
 #define safe_object_get_size	sgen_safe_object_get_size
 
+#if defined(TARGET_OSX) || defined(TARGET_WIN32) || (defined(__linux__) && !defined(TARGET_ANDROID))
```

Comment:
* `PLATFORM_MACOSX`: https://github.com/mono/mono/blob/master/configure.ac#L283
* `PLATFORM_LINUX`: https://github.com/mono/mono/blob/master/configure.ac#L350

Although in the case of `PLATFORM_LINUX`, we are for some reason looking at `target_os`. Not terribly hard to fix at a later point, though. It shouldn't do any harm to use the macro now anyhow.